api/end_device.proto

```diff
@@ -429,11 +429,11 @@ message EndDevice {
   // The power state of the device; whether it is battery-powered or connected to an external power source.
   // Received via the DevStatus MAC command at status_received_at.
   // Stored in Network Server.
-  PowerState power_state = 34;
+  PowerState power_state = 34 [(validate.rules).enum.defined_only = true];
   // Latest-known battery percentage of the device.
   // Received via the DevStatus MAC command at last_dev_status_received_at or earlier.
   // Stored in Network Server.
-  float battery_percentage = 35;
+  float battery_percentage = 35 [(validate.rules).float = {gte: -1, lte:1}];
```

Comment:
Is `-0.5` a valid value?

Also spacing

Reply:
1. No, but I did not find a better way to allow `[-1]∪[0,1]`. NS uses `-1` when it's not known.
2. What's wrong with spacing?

Reply:
1. Shouldn't this be nullable then?
2. Between `lte` and `1`

Reply:
Indeed, will change this to `FloatValue`